mev_inspect/inspector.py

```diff
@@ -0,0 +1,82 @@
+import asyncio
+import logging
+import sys
+import traceback
+from asyncio import CancelledError
+
+from web3 import Web3
+from web3.eth import AsyncEth
+
+from mev_inspect.block import create_from_block_number
+from mev_inspect.classifiers.trace import TraceClassifier
+from mev_inspect.db import get_inspect_session, get_trace_session
+from mev_inspect.inspect_block import inspect_block
+from mev_inspect.provider import get_base_provider
+
+logging.basicConfig(stream=sys.stdout, level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+class MEVInspector:
+    def __init__(
+        self,
+        rpc: str,
+        cache: bool = False,
+        max_concurrency: int = 1,
+        request_timeout: int = 300,
+    ):
+        if not cache:
+            logger.info("Skipping cache")
+
+        self.inspect_db_session = get_inspect_session()
+        self.trace_db_session = get_trace_session()
```

Comment:
Well in the end the sessions get created only once since the class itself is instantiated once only upon instantiation -- that same session is then passed to the underlying tasks. 

I guess it becomes a matter of preference, I usually like the cli code being kept as minimal as possible, and just assembling the necessary dependencies on the Inspector class. Only issue I see is that the inspector class might become a bit inflexible in case more functionality needs to be added

Reply:
I'm cool with that. Let's keep it in here and can feel it out as we go